scipy/sparse/construct.py

```diff
@@ -644,38 +645,73 @@ def block_diag(mats, format=None, dtype=None):
 
     Notes
     -----
+    Providing a sequence of equally shaped matrices
+     will provide marginally faster results
 
-    .. versionadded:: 0.11.0
+    .. versionadded:: 0.18.0
 
     See Also
     --------
-    bmat, diags
+    bmat, diags, block_diag
 
     Examples
     --------
     >>> from scipy.sparse import coo_matrix, block_diag
     >>> A = coo_matrix([[1, 2], [3, 4]])
-    >>> B = coo_matrix([[5], [6]])
-    >>> C = coo_matrix([[7]])
+    >>> B = coo_matrix([[5, 6], [7, 8]])
+    >>> C = coo_matrix([[9, 10], [11,12]])
     >>> block_diag((A, B, C)).toarray()
-    array([[1, 2, 0, 0],
-           [3, 4, 0, 0],
-           [0, 0, 5, 0],
-           [0, 0, 6, 0],
-           [0, 0, 0, 7]])
-
+    array([[ 1,  2,  0,  0,  0,  0],
+           [ 3,  4,  0,  0,  0,  0],
+           [ 0,  0,  5,  6,  0,  0],
+           [ 0,  0,  7,  8,  0,  0],
+           [ 0,  0,  0,  0,  9, 10],
+           [ 0,  0,  0,  0, 11, 12]])
```

Comment:
This example doesn't need to be updated, as the original was fine for demonstrating the capabilities of the function.